service/storage_service.cc

```diff
@@ -6885,6 +6885,33 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
     });
 }
 
+future<locator::load_stats> storage_service::load_stats_for_tablet_based_tables() {
+    auto holder = _async_gate.hold();
+
+    if (this_shard_id() != 0) {
+        // group0 is only set on shard 0.
+        co_return co_await container().invoke_on(0, [&] (auto& ss) {
+            return ss.load_stats_for_tablet_based_tables();
+        });
+    }
+
+    auto load_stats = co_await _db.map_reduce0([] (replica::database& db) mutable -> future<locator::load_stats> {
+        locator::load_stats load_stats{};
+        auto& tables_metadata = db.get_tables_metadata();
+
+        co_await tables_metadata.for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) {
+            if (table->uses_tablets()) {
+                load_stats.tables.emplace(id, table->table_load_stats());
+            }
+            return make_ready_future<>();
+        });
+
+        co_return std::move(load_stats);
+    }, locator::load_stats{}, std::plus<locator::load_stats>());
```

Comment:
Suggest to extract the inner into a named function. Then you can call map_reduce0(std::mem_fn(...), locator::load_stats{}, std::plus<>()).

Reply:
Ok.

Reply:
forgotten?